### PR TITLE
Fix xgboost cyclic and move codes into app_opt [skip ci]

### PR DESCRIPTION
**Pull request by YuanTingHsieh**
- Fix XGBoost tree-based cyclic algo config
- Move codes into app_opt/xgboost for consistency
- Update readme and configs

### CODE REVIEW

File: examples/xgboost/README.md [hunk 1]
```diff
@@ -87,31 +89,25 @@ We then prepare the job configs for NVFlare jobs corresponding to various settin
...
+- tree-based bagging training with non-uniform data split and scaled shrinkage for 5/20 clients
 
-For gpu based training, edit `job_conf_gen.sh` to change `TREE_METHOD="hist"` to `TREE_METHOD="gpu_hist"`.
 
-> **_NOTE:_** Cyclic training always use uniform shrinkage. The generated job config files will be stored in the folder `./job_configs`
+The script will also generate 2 configs in `histogram-based/job_configs` for histogram-base algorithm:
 
-### Installation
-Follow the [Installation](https://nvflare.readthedocs.io/en/main/quickstart.html) instructions.
+- histogram-based training with uniform data split for 2/5 clients
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply examples/xgboost/README.md:108]: for 2 or for 5 clients

File: examples/xgboost/histogram-based/README.md [hunk 2]
```diff
@@ -5,18 +5,18 @@ To run this example with NVFlare, follow the below steps.
...
 ```
 bash run_experiment_simulator.sh
 ```
 
-## Results on 5- and 20-client under various training settings
-For comparison, we train baseline models in a centralized manner with same round of training
+## Centralized training
 ```
 bash run_experiment_centralized.sh
 ```
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply examples/xgboost/histogram-based/README.md:22]: added

**Review by ZiyueXu77 — APPROVED** [review]
LGTM

**Pull request merged**